File: backend/Domain/Entities/Customers/Customer.cs
using Domain.Entities.Projects;
using Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities.Customers
{
    public class Customer : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? Company { get; set; }
        public string? Address { get; set; }
        public CustomerStatus Status { get; set; } = CustomerStatus.Lead;
        public string? Notes { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }


        public ICollection<Project> Projects { get; set; }
            = new List<Project>();
    }
}

File: Frontend/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  email?: string | null;
  phone?: string | null;
  company?: string | null;
  address?: string | null;
  status: string;
  notes?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CustomerRequest {
  name: string;
  email?: string | null;
  phone?: string | null;
  company?: string | null;
  address?: string | null;
  status: string;
  notes?: string | null;
}

export interface CustomerPagedResponse {
  items: Customer[];
  page: number;
  pageSize: number;
  totalItems: number;
  totalPages: number;
}
